public/app/features/api-keys/ApiKeysPage.tsx

```diff
@@ -146,10 +150,81 @@ export class ApiKeysPageUnconnected extends PureComponent<Props, State> {
             ) : null}
           </>
         </Page.Contents>
+        <MigrationSummary
+          visible={this.state.migrationSummaryVisible}
+          data={migrationResult}
+          dismissModal={this.dismissModal}
+        />
       </Page>
     );
   }
 }
+export type MigrationSummaryProps = {
+  visible: boolean;
+  data: ApikeyMigrationResult;
+  dismissModal: () => void;
```

Comment:
It's better to use the same name convention. This is a callback that executed when you dismiss the modal, so onDismiss is a good name. Usually name pattern for the callbacks in react is `on<Action>`
```suggestion
  onDismiss: () => void;
```